plugins/woocommerce/client/admin/client/settings-payments/components/country-selector/country-selector.tsx

```diff
@@ -0,0 +1,308 @@
+			// without circular navigation. Otherwise, select the first item.
+			return {
+				selectedItem:
+					items[
+						selectedItem
+							? Math.min(
+									items.indexOf( selectedItem ) + 1,
+									items.length - 1
+							  )
+							: 0
+					],
+				isOpen: true, // Keep menu open after selection.
+			};
+		case useSelect.stateChangeTypes.ToggleButtonKeyDownArrowUp:
+			// If we already have a selected item, try to select the previous one,
+			// without circular navigation. Otherwise, select the last item.
+			return {
+				selectedItem:
+					items[
+						selectedItem
+							? Math.max( items.indexOf( selectedItem ) - 1, 0 )
+							: items.length - 1
+					],
+				isOpen: true, // Keep menu open after selection.
+			};
+		default:
+			return changes;
+	}
+};
+
+const removeAccents = ( str: string ) => {
+	return str.normalize( 'NFD' ).replace( /[\u0300-\u036f]/g, '' );
+};
+
+export const CountrySelector = < ItemType extends Item >( {
+	name,
+	className,
+	label,
+	describedBy,
+	options: items,
+	onChange,
+	value,
+	placeholder,
+	children,
+}: ControlProps< ItemType > ): JSX.Element => {
+	const [ searchText, setSearchText ] = useState( '' );
+	const [ isSearchFocused, setSearchFocused ] = useState( false );
+
+	// only run filter every 200ms even if the user is typing
+	const throttledApplySearchToItems = useThrottle(
+		useCallback(
+			( searchString: string, itemSet: ItemType[] ) =>
+				new Set(
+					itemSet.filter( ( item: Item ) =>
+						`${ removeAccents( item.name ?? '' ) }`
+							.toLowerCase()
+							.includes( removeAccents( searchString ) )
+					)
+				),
+			[]
+		),
+		200
+	);
+
+	const visibleItems =
+		searchText !== ''
+			? throttledApplySearchToItems( searchText, items ) ?? new Set()
+			: new Set( items );
+
+	const {
+		getToggleButtonProps,
+		getMenuProps,
+		getItemProps,
+		isOpen,
+		highlightedIndex,
+		selectedItem,
+		closeMenu,
+		selectItem,
+	} = useSelect< ItemType >( {
+		initialSelectedItem: value,
+		items: [ ...visibleItems ],
+		stateReducer,
+		onIsOpenChange: () => selectItem( value ),
+	} );
+
+	const itemString = getOptionLabel( value.key, items );
+	const selectedValue = selectedItem ? selectedItem.key : '';
+
+	const searchRef = useRef< HTMLInputElement >( null );
+	function getDescribedBy() {
+		if ( describedBy ) {
+			return describedBy;
+		}
+
+		if ( ! itemString ) {
+			return __( 'No selection', 'woocommerce' );
+		}
+
+		return sprintf(
+			// translators: %s: The selected option.
+			__( 'Currently selected: %s', 'woocommerce' ),
+			itemString
+		);
+	}
+
+	const getSearchSuffix = ( focused: boolean ) => {
+		if ( focused ) {
+			return (
+				<img
+					src={
+						WC_ASSET_URL + 'images/settings-payments/clear-icon.svg'
+					}
+					alt={ __( 'Clear search', 'woocommerce' ) }
+				/>
+			);
+		}
+
+		return (
+			<img
+				src={
+					WC_ASSET_URL + 'images/settings-payments/search-icon.svg'
+				}
+				alt={ __( 'Clear search', 'woocommerce' ) }
```

Comment:
probably should be `Search`

Reply:
Are you referring to the icon or the alt text?

Reply:
alt text!

Reply:
I see what you mean, the alt text for both the search and clear icons was the same. It’s now fixed.